# Tools for Bioinformatics
A final portfolio for the "Bioinformatics: Tools for Genome Analysis (AS.410.635.81)" course from the JHU Individualized Genomics and Health program.
This portfolio contains examples of experience with various bioinformatic tools, of lessons learned and discussions partaken in.
Sections have been split in this repository relating to examples of different tools used during the BTGA course and others
such as "Next Generation DNA Sequencing Analysis" and "Genomic Expression Data Analysis and Visualization".
*Each file on GitHub has a .txt file that describes each in detail.*

## [Coursework from Bioinformatics: Tools for Genome Analysis](https://github.com/agilson2/AGilson-Portfolio/tree/main/BTGA%20Classwork)
* Both reading summaries and discussion posts from throughout the spring 2021 semester in Bioinformatics: Tools for Genome Analysis
* Reading summaries were completed every week to synthesize the most important points from each week's lecture. 
  * These are broken up by individual units.
  * These units all contained information about tools that were later applied to the research paper and assingments.
* Class Discussions were in relation to the unit's subject each week, but went beyond the base level of the lectures to spark interesting conversation.

## [Galaxy and Other Bioinformatic Tools](https://github.com/agilson2/AGilson-Portfolio/tree/main/Galaxy%20and%20Other%20Tools) 
* Contains multiple different examples from various tools.
* Most folders contain supplementary files that relate to the data presented in the file.

### [Galaxy](https://github.com/agilson2/AGilson-Portfolio/tree/main/Galaxy%20and%20Other%20Tools/Galaxy)
* Galaxy is an open source platform in which users can more easily manipulate bioinformatic data, found [here](https://usegalaxy.org/).
* Through this platform, I was exposed to a variety of different file formats including BED, WIG, BAM, SAM, and many others. 
* The main focus of using Galaxy in the BTGA course was to facilitate the analysis of larger datasets.

### [Genome Browsers](https://github.com/agilson2/AGilson-Portfolio/tree/main/Galaxy%20and%20Other%20Tools/Genome%20Browsers)
* A genome browser presents a unique visual and graphical display of information from a dataset that wouldn't otherwise be seen. Examples of browsers used include
* Integrative Genomics Viewer
* NCBI Genome Browser
* UCSC Genome Browser
* Ensembl

### [Glimmer and Other Command Line Tools](https://github.com/agilson2/AGilson-Portfolio/tree/main/Galaxy%20and%20Other%20Tools/Glimmer%20%26%20Command%20Line)
* The BFX3 server available on the JHU network was essential in accessing many command line tools, such as glimmer.
* Glimmer is a system of predicting gene loci in bacteria, archaea, and viruses. 
* Other tools include
  * BEDtools: allows data manipulation and analysis for information found in BED files
  * SRAtoolkit: allows for the manipulation and analysis of raw sequencing reads presented by the sequencing read archive(SRA).

### [Prediction Software](https://github.com/agilson2/AGilson-Portfolio/tree/main/Galaxy%20and%20Other%20Tools/Prediction%20Software)
* Along with the other analysis tools, a large part of the work done in courses throughout my master's program involved the use of programs that predicted outcomes
* Simple use of FASTA and BLAST was utilized throughout each course in the program
* For the research paper on the SNP rs75932146, many protein structure, stability, and functional change programs were utilized. These included: COACH, DynaMut, mCSM, SIFT, SDM, and many more. Please see the PDF for more information on these programs.

### [R](https://github.com/agilson2/AGilson-Portfolio/tree/main/Galaxy%20and%20Other%20Tools/R)
* R is an incredible tool for bioinformatic analysis. 
* It was used in my own research/presentation of age related changes in gene expression between the dermis and epidermis.
  * *During the course Gene Expression Data Analysis and Visualization*
* It was also used to pull data from the Biomart genomic database hosted by Ensembl via the biomaRt package. 


